### PR TITLE
[eslint-config] Add new plugins

### LINKED ISSUE #58: [eslint-plugin] 신규 플러그인 / 룰 검토
### O

- [x] eslint-config-standard 제거
  - neostandards 로 대체
- [x] eslint-plugin-yml 
  - 현재 yaml은 prettier 에 위임하고 있지만,
  - fomatting -> prettier, 외에 룰 도입

#### strict config 추가

- [x] [eslint-plugin-sonarjs](https://github.com/SonarSource/SonarJS/blob/master/packages/jsts/src/rules/README.md) 보류
  - 신규 프로젝트에는 괜찮을듯하지만, 기존 프로젝트에는 파급력이 커보임
  - `eslint-plugin-unicorn`랑 충돌나는 옵션 고려 필요
- [x] [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) 보류
  - 위와 동일한 이유

---

### X

- [x] eslint-plugin-node 대신 eslint-plugin-n 으로 (업데이트가 느려터짐)
  - eslint-plugin-node 제거 됌
- [x] [eslint-plugin-workspaces](https://github.com/joshuajaco/eslint-plugin-workspaces) 보류
  - 현재 코드에 변경사항이 크지 않다면 적용해볼만함
  - 워크스페이스 아닌 환경 고려
- [x] eslint-plugin-compat 보류
  - 플러그인은 맘에 들지만 browserlist 기반으로 동작해, 사용하는 쪽에서 별도의 설정이 필요하다는 내용을 명시해줘야함
- [ ] [eslint-plugin-vitest](https://github.com/vitest-dev/eslint-plugin-vitest) 보류
    제공자 입장에서 대상 파일을 선언하는 방식은 지양하기 위해 보류
    ```js
    export default [
        js.configs.recommended,
        {
            files: ['**/__tests__/**', '**/tests/**', '**/*{.,_}{test,spec}.{js,cjs,mjs,jsx,ts,cts,mts,tsx}'],
            ...vitest.configs.recommended,
        },
    ]
    ```
- [x] [eslint-plugin-css-modules](https://github.com/atfzl/eslint-plugin-css-modules)
  - Project status: NOT MAINTAINED
- [x] [eslint-plugin-deprecation](https://github.com/gund/eslint-plugin-deprecation)
-> deprecated // https://typescript-eslint.io/rules/no-deprecated
n/no-deprecated-api 가 역할을 하고 있는 것으로 보임

### PR DESCRIPTION
**Pull request by yongholeeme**
## Related Issue

close #58

## Describe your changes

- Add [eslint-plugin-yml](https://www.npmjs.com/package/eslint-plugin-yml) to node, react
- Enforce a convention in the order to node
- Disallow unused imports to node, react
- Enforce consistent usage of type imports to react/typescript
  -  Add a recommend [`typescript.preferences.preferTypeOnlyAutoImports`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-3/#settings-to-prefer-type-auto-imports)
- Add strict config
  - extends [eslint-plugin-unicorn](https://www.npmjs.com/package/eslint-plugin-unicorn), [eslint-config-sonarjs](https://www.npmjs.com/package/eslint-plugin-sonarjs)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,17 +13,17 @@
         "prelint": "pnpm run build",
         "lint": "eslint",
         "lint:fix": "pnpm run lint --fix",
-        "prettier": "prettier --check '**/*.{json,yaml,md,ts,tsx,js,jsx}'",
-        "prettier:fix": "prettier --write '**/*.{json,yaml,md,ts,tsx,js,jsx}'",
+        "prettier": "prettier --check '**/*.{json,yaml,yml,md,ts,tsx,js,jsx}'",
+        "prettier:fix": "prettier --write '**/*.{json,yaml,yml,md,ts,tsx,js,jsx}'",
```
Comment (yceffort-naver) [inline package.json:17]: 요것도 breaking change 다
의견 반영감사합니다

**Review by yceffort-naver — APPROVED** [review]
좋다 좋다~~

**Review by yujeong-jeon — APPROVED** [review]
룰 업뎃 감사드립니다 :D 👍

File: packages/eslint-config/react/configs.js [hunk 2]
```diff
@@ -18,4 +20,16 @@ export default [
...
+    /**
+     * Disable rules that cause problems with YAML files requiring type information
+     */
+    {
+        files: ['**/*.yaml', '**/*.yml'],
+        rules: {
+            '@typescript-eslint/naming-convention': 'off',
+            '@typescript-eslint/consistent-type-imports': 'off',
+        },
+    },
```
Comment (yongholeeme) [inline packages/eslint-config/react/configs.js:34]: 😭

**Review by 2-one-week — APPROVED** [review]
고생 많으셨습니다

File: packages/eslint-config/react/configs.js (hunk 2 above)
Comment (yongholeeme) [reply packages/eslint-config/react/configs.js:34]: dㅣ거 조금 걱정인데요, 저희 config 사용하면서 `@typescript-eslint` 룰 중에 타입 체킹관련 룰 추가하면 저 설정 추가해줘야하는데 관련 내용을 명시해줘야겠죠?;

Comment (yceffort-naver) [reply packages/eslint-config/react/configs.js:34]: warning 은 써줘야할듯.. README에

**Pull request merged**